### PR TITLE
feat: hammer sync example

### PR DESCRIPTION
**Pull request by Arqu**
## Change checklist

- [x] Self-review.
- [ ] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

**Comment by Arqu** [comment]
Dont merge yet, working on some extensions

**Comment by dignifiedquire** [comment]
clippy sad

File: iroh/examples/sync.rs [hunk 1]
```diff
@@ -502,6 +546,17 @@ pub enum Cmd {
     WatchCancel,
     /// Show stats about the current session
     Stats,
+    /// Stress test with the hammer
+    Hammer {
```
Comment (Frando) [inline iroh/examples/sync.rs:550]: Now that we're at four arguments, I'd prefer to turn them into options. Then you can also add default values if needed.
`#[clap(short, long)]` etc

File: iroh/examples/sync.rs [hunk 2]
```diff
@@ -349,6 +349,50 @@ async fn handle_command(
...
+        } => {
+            println!(
+                "> Hammering with prefix {prefix} for {threads} x {count} messages of size {size} bytes",
+                prefix = prefix,
+                threads = threads,
+                count = count,
+                size = size,
+            );
+            let mut bytes = vec![0; size];
+            bytes.fill(97);
```
Comment (Frando) [inline iroh/examples/sync.rs:366]: If we also want to test content downloading with this (do we?) maybe let's set their content to an increasing number or add an option for that behavior. Can also come later though.

**Pull request merged**